### PR TITLE
Addition of NewImageFromURI

### PR DESCRIPTION
**Pull request by andydotxyz**
### Description:
A discussion point reall. `NewImageFromFile` isn't really inline with the new `URI` based file access - so this seems sensible.
However should the `http` and `https` scheme support be in the driver, or the image code or somewhere else?
Basically: is `OpenFileAtURI` file specific or not - and if yes should there be another method call for non-files?

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.
- [x] Public APIs match existing style.

### CODE REVIEW

File: canvas/image.go [hunk 1]
```diff
@@ -74,6 +79,49 @@ func NewImageFromFile(file string) *Image {
...
+// Images returned from this method will scale to fit the canvas object.
+// The method for scaling can be set using the Fill field.
+func NewImageFromURI(uri fyne.URI) *Image {
+	if uri.Scheme() == "file" && len(uri.String()) > 7 {
+		return &Image{
+			File: uri.String()[7:],
+		}
+	}
+
+	var read io.ReadCloser
```
Comment (stuartmscott) [inline canvas/image.go:93]: Could this be split out into `NewImageFromReader`?

Comment (andydotxyz) [reply canvas/image.go:93]: A cunning plan. Does the rest of the PR seem sensible? I was more keen to get feedback on the http/https part in the  description.

**Review by Jacalz — COMMENTED** [review]
This is a great addition. My view on this is that the http/https probably belongs in the driver (unless the go stdlib has us covered in all situations). I also think a general URI method makes sense.

**Comment by andydotxyz** [comment]
I don't think that every driver needs to define how http works - that is built into Go.
The uncertainty is because this would be the only usage of URI where that would work.
There is a discussion about how the toolkit (and developers) could provide URI scheme handling through somesort of register, the http code may belong in there and we could provide this scheme support universally.

**Comment by Jacalz** [comment]
I agree. That sounds like a good plan to me.

File: canvas/image.go (hunk 1 above)
Comment (andydotxyz) [reply canvas/image.go:105]: Added

**Comment by andydotxyz** [comment]
OK, I have updated this PR to remove the HTTP/HTTPS handling as that belongs in the discussed URI handler registry.
Otherwise updated as per comments.

**Review by stuartmscott — APPROVED** [review]

**Review by stuartmscott — APPROVED** [review]

**Pull request merged**